src/Application/Factories/AlarmMonitoredItemServiceInitializerFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿// SPDX-License-Identifier: MIT. 
+// Copyright Contributors to the Open Manufacturing Platform.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using OMP.Connector.Domain;
+using OMP.Connector.Domain.Models.Telemetry;
+using OMP.Connector.Domain.OpcUa.Services;
+using OMP.Connector.Domain.Schema;
+using Opc.Ua.Client;
+
+namespace OMP.Connector.Application.Factories
+{
+    public class AlarmMonitoredItemServiceInitializerFactory : IAlarmMonitoredItemServiceInitializerFactory
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public AlarmMonitoredItemServiceInitializerFactory(IServiceProvider serviceProvider)
```

Comment:
Why do we inject the DI container here? By doing this we do not clearly state to the outside world what is required for this class to function, so the outside world needs to know what is going on inside to be able to use this class. Also problematic for unit test.
Is there an edge case here regarding the timing of external initialization why we do this?

Reply:
I agree and I cannot recall the reason why it was implemented this way in the Subscription. 
This pattern was copied from the normal Subscription and it existed before the code was migrated to OMP.

I have removed the unnecessary complexity with the dependency injection.